pages/documentation/simulation/sensitivity/openlf.md

```diff
@@ -198,6 +199,192 @@ In that case, the vector of right-hand side $$b_1$$ is the injections at network
+5. Compute $$J(v,\phi)^{-1}f_p(v,\phi)$$ with the LU decomposition ;
+6. Compute $$G_{v,\phi}(v,\phi)$$ ;
+7. Compute $$g_p(v,\phi)$$ ;
+8. Compute $$S_{\eta,p}(v,\phi)$$ using the formula.
+
+Following sections detail how $$f_p(v,\phi)$$, $$G_{v,\phi}(v,\phi)$$ and $$g_p(v,\phi)$$ are computed.
+
+### Computation of vector $$f_p(v,\phi)$$
+
+Vector $$f_p(v,\phi)$$ is the gradient of power flow equations according to the parameter $$p$$, at the point $$(v,\phi)$$. Its computation depends on if $$p$$ is the active injection at a bus, or the phase shifting angle of a phase tap changer.
+
+#### Case 1: $$p$$ is the active injection at bus $$i$$
+
+In this case, all entries of vector $$f_p(v,\phi)$$ equal zero, except one which corresponds to the active power flow balance equation at bus $$i$$:
+
+$$
+\begin{align}
+\texttt{If}~k~\texttt{is the active power flow balance at bus}~i:& \quad (f_p(v,\phi))_k = 1,\\
+\\
+\texttt{Else}:& \quad (f_p(v,\phi))_k = 0.
+\end{align}
+$$
+
+#### Case 2: $$p$$ is the phase shifting angle of the phase tap changer on side $$i$$ of line $$(i,j)$$
+
+In this case, all entries of vector $$f_p(v,\phi)$$ equal zero, except following ones:
+- active power flow balance eqaution at both buses $$i$$ and $$j$$,
+- reactive power flow balance equation at bus $$i$$, only if it is a PQ-bus,
+- reactive power flow balance equation at bus $$j$$, only if it is a PQ-bus.
+
+$$
+\begin{align}
+\texttt{Let}~\alpha &= -\rho_iv_iY\rho_jv_j\text{cos}(\theta)\frac{\pi}{180},\\
+\\
+\texttt{Let}~\beta &= \rho_iv_iY\rho_jv_j\text{sin}(\theta)\frac{\pi}{180}.
+\end{align}
+$$
+
+$$
+\begin{align}
+\texttt{If}~k~\texttt{is the active power flow balance at bus}~i:& \quad (f_p(v,\phi))_k = \alpha,\\
+\\
+\texttt{If}~k~\texttt{is the active power flow balance at bus}~j:& \quad (f_p(v,\phi))_k = -\alpha,\\
+\\
+\texttt{If}~k~\texttt{is the reactive power flow balance at bus}~i~\texttt{which is PQ}:& \quad (f_p(v,\phi))_k = \beta,\\
+\\
+\texttt{If}~k~\texttt{is the reactive power flow balance at bus}~j~\texttt{which is PQ}:& \quad (f_p(v,\phi))_k = -\beta,\\
+\\
+\texttt{Else}:& \quad (f_p(v,\phi))_k = 0.
+\end{align}
+$$
+
+### Computation of matrix $$G_{v,\phi}(v,\phi)$$
+
+Matrix $$G_{v,\phi}(v,\phi)$$ is the matrix whose each column is the gradient of a sensitivity according to state variables $$v$$ and $$\phi$$, at the point $$(v,\phi)$$. It is computed column by column. The computation of column $$l$$ depends on if it is relative to the sensitivity of the power or the current flowing from $$i$$ to $$j$$.
+
+#### Case 1: Column $$l$$ is relative to the sensitivity of the power flowing from $$i$$ to $$j$$
+
+In this case, all entries of column $$l$$ of $$G_{v,\phi}(v,\phi)$$ equal zero, except four which correspond to the voltage magnitudes and angles at both buses $$i$$ and $$j$$:
+
+$$
+\begin{align}
+\texttt{Let}~\frac{dp}{dv_i} &= \rho_i(2G\rho_iv_i + 2Y\rho_iv_i\text{sin}(\Xi) - Y\rho_jv_j\text{sin}(\theta)),\\
```

Comment:
This is `dp1` ?

Reply:
This is dp1/dv1, which is contained in variable dp1dv1 in the code.